### ISSUE TITLE
`atomicCompareExchangeWeak` is processed incorrectly by `compose`.

### ISSUE THREAD

**Issue by Joeoc2001** [bevyengine/naga_oil#48]
Issue: When using this crate to parse a wgsl file containing an `atomicCompareExchangeWeak` instruction, the second argument and result type are switched with other types and expressions in scope when the call is performed. 

Versions: (all latest at time of writing)
```
naga_oil = "0.9"
naga = { version = "0.13", features = ["wgsl-in", "wgsl-out"] }
```

Shader (wgsl):
```wgsl
@group(0) @binding(0) var<storage, read_write> foo: atomic<u32>;

@compute
@workgroup_size(256,1,1)
fn bar(@builtin(global_invocation_id) global_id: vec3<u32>) {
    atomicCompareExchangeWeak(&foo, 0u, global_id.x);
}
```

Rust:
```rust
fn main() {
    let mut composer = naga_oil::compose::Composer::non_validating();
    let module = composer
        .make_naga_module(NagaModuleDescriptor {
            source: include_str!("main.wgsl"),
            file_path: "main.wgsl",
            shader_type: naga_oil::compose::ShaderType::Wgsl,
            shader_defs: HashMap::default(),
            additional_imports: &[],
        })
        .unwrap();

    let info = naga::valid::Validator::new(
        naga::valid::ValidationFlags::empty(),
        naga::valid::Capabilities::empty(),
    )
    .validate(&module)
    .unwrap();

    let res = naga::back::wgsl::write_string(
        &module,
        &info,
        naga::back::wgsl::WriterFlags::EXPLICIT_TYPES,
    )
    .unwrap();

    println!("{}", res);
}
```

Result:
```wgsl
struct gen___atomic_compare_exchange_resultUint4_ {
    old_value: u32,
    exchanged: bool,
}

@group(0) @binding(0) 
var<storage, read_write> foo: atomic<u32>;

@compute @workgroup_size(256, 1, 1) 
fn bar(@builtin(global_invocation_id) global_id: vec3<u32>) {
    let _e3: vec3<u32> = atomicCompareExchangeWeak((&foo), (&foo), global_id.x);
    return;
}
```

Note that both the type of `_e3`, and the second argument of `atomicCompareExchangeWeak`, are incorrect.

Performing a parse and write cycle with plain `naga` does not have this issue, so the type and argument switching occurs in this crate.

**Comment by robtfm** [comment]
Thanks for the report - ill have time to look properly in a couple of days, but if you’d like to try a possible fix, adding `is_external = true;` to the atomic case [here](https://github.com/bevyengine/naga_oil/blob/185456741f5e4843f35a145905a57c9056385bc0/src/derive.rs#L571) might do the trick

**Comment by Joeoc2001** [comment]
This was fixed by PR #49
